backend/core/src/util/hash.rs

```diff
@@ -55,6 +63,20 @@ impl<'de> Deserialize<'de> for Hash {
     }
 }
 
+impl Serialize for Hash {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut ascii = [0u8; 2 + HASH_BYTES * 2];
+
+        ascii[..2].copy_from_slice(b"0x");
+
+        hex::encode_to_slice(&self.0, &mut ascii[2..]).expect("Encoding 32 bytes into 64 bytes of ascii; qe");
+
+        serializer.serialize_str(
+            std::str::from_utf8(&ascii).expect("Hex encoding is always valid utf8; qed"),
```

Comment:
Maybe we can even be bold here and go with `from_utf8_unchecked()`? After all `from_utf8` + `expect` is sort of the same thing?

Reply:
The unchecked version is `unsafe` iirc, so this one would be safer regardless :)